File: Day_4/happy_ladybugs.js
// Complete the happyLadybugs function below.
function happyLadybugs(b) {
    var status = "YES";
    var tempArr = new Array(96).fill(0);
    for (let i = 0; i < b.length; i++){
        tempArr[b.charCodeAt(i)]++;
    }
    for (let i = 1; i < b.length - 1; i++){
        if (b[i] != b[i + 1] && b[i] != b[i - 1] && tempArr[95] < 1) {
            status = "NO";
        }
    }
    for (let i = 65; i <= 90; i++){
        if (b.indexOf(String.fromCodePoint(i)) != -1 && tempArr[i] < 2) {
            status = "NO";
        }
    }
    return status;
}

File: Day_4/happy_ladybugs.java
 static String happyLadybugs(String s) {
        boolean hasUnderscore = false;
        boolean allAreHappy = true;
        int countArray[] = new int[26];

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '_') {
                hasUnderscore = true;
            } else {
                countArray[s.charAt(i) - 'A']++;
                if (( i > 0 && s.charAt(i-1) == s.charAt(i)) || (i+1 < s.length() && s.charAt(i+1) == s.charAt(i))) {
                } else {
                    allAreHappy = false;
                }
            }
        }
        
        if (allAreHappy) {
            return "YES";
        }
        
        if (!hasUnderscore) {
            return "NO";
        }
        
        for (int i = 0; i < countArray.length; i++) {
            if (countArray[i] == 1) {
                return "NO";
            }
        }
        
        return "YES";
    }